### PR TITLE
Cherry-pick #24 and #26 into legacy-agent-rpm.

### PR DESCRIPTION
**Pull request by dhrupadb**

### CODE REVIEW

File: stackdriver-agent [hunk 1]
```diff
@@ -62,104 +61,36 @@ if [ -n "$JAVA_LIB_DIR" ]; then
...
+    if ! $DAEMON -t -C "$CONFIG"; then
+        return 1
     fi
-    if [ -z "$COLLECTD_ENDPOINT"]; then
-        COLLECTD_ENDPOINT=$(get_collectd_endpoint)
-        if [[ $? -ne 0 ]]; then
-            return 1
+    # Check if the application default credentials file is in the system
+    # location.
+    if [ ! -f "${GOOGLE_APPLICATION_CREDENTIALS:-/etc/google/auth/application_default_credentials.json}" ]; then
```
Comment (igorpeshansky) [inline stackdriver-agent]: Looks like you're also picking up #27.

**Comment by qingling128** [comment]
NOTE per discussion offline: This might have broken the Amazon build. Holding on merging before further investigation.

**Comment by qingling128** [comment]
The last commit fixed the build. @igorpeshansky - Mind taking a quick look at https://github.com/Stackdriver/agent-packaging/pull/33/commits/042f843df89c2d9137b1c11012d528e1e8b49e2d before I merge this PR?

**Comment by qingling128** [comment]
BTW, #27 was cherry-picked in https://github.com/Stackdriver/agent-packaging/pull/29 already. So this is only about #24 and #26. I just adjusted the title.

File: stackdriver-agent (hunk 1 above)
Comment (qingling128) [reply stackdriver-agent]: Just double checked that #27 was cherry-picked already by https://github.com/Stackdriver/agent-packaging/pull/29.

**Comment by igorpeshansky** [comment]
dhrupadb was at Google at the time.

**Pull request merged**